File: 0352-data-stream-as-disjoint-intervals/0352-data-stream-as-disjoint-intervals.js

var SummaryRanges = function() {
    this.num = []
};

/** 
 * @param {number} val
 * @return {void}
 */
SummaryRanges.prototype.addNum = function(val) {
    if (this.num.includes(val)) return 
    this.num.push(val)
    this.num.sort((a, b) => a - b)
};

/**
 * @return {number[][]}
 */
SummaryRanges.prototype.getIntervals = function() {
    const area = []
    const n = this.num.length
    if (!n) return area

    for (let i = 0; i < n; i++) {
        const curr = this.num[i]
        if (!area.length) area.push([curr, curr])

        else {
            const lastArea = area[area.length - 1]
            if (lastArea[1] + 1 === curr)
                lastArea[1]++
            
            else area.push([curr, curr])
        }
    }

    return area
};

/** 
 * Your SummaryRanges object will be instantiated and called as such:
 * var obj = new SummaryRanges()
 * obj.addNum(value)
 * var param_2 = obj.getIntervals()
 */

File: 0352-data-stream-as-disjoint-intervals/0352-data-stream-as-disjoint-intervals.py
class SummaryRanges:

    def __init__(self):
        self.nums = []

    def addNum(self, value: int) -> None:
        if value in self.nums:
            return
        self.nums.append(value)
        self.nums.sort()

    def getIntervals(self) -> List[List[int]]:
        area = []
        n = len(self.nums)
        if n == 0:
            return area
        for i, num in enumerate(self.nums):
            if not area:
                area.append([num, num])
            else:
                lastArea = area[-1]
                if lastArea[1] + 1 == num:
                    lastArea[1] += 1
                else:
                    area.append([num, num])
        return area
            


# Your SummaryRanges object will be instantiated and called as such:
# obj = SummaryRanges()
# obj.addNum(value)
# param_2 = obj.getIntervals()